### PR TITLE
refactor(rust-sdk): async APIs return JSON string.

### PR DESCRIPTION
**Pull request by agustinustheo**
### Changelog
- Public and private async APIs now return results in the form of a JSON string.
- Update integration tests to check if the error Results are empty.

### CODE REVIEW

File: rust/src/main.rs [hunk 1]
```diff
@@ -9,134 +9,38 @@ async fn async_main() {
...
-        Err(e) => println!("{:?}", e),
-        _ => ()
-    }
+    println!("{:?}", public::get_server_time_async().await);
+    println!("{:?}", public::get_markets_async().await);
+    println!("{:?}", public::get_products_async().await);
+    println!("{:?}", public::get_assets_async().await);
+    println!("{:?}", public::get_crypto_assets_async().await);
+    println!("{:?}", public::get_fiat_assets_async().await);
+    println!("{:?}", public::get_asset_pairs_async("crypto","spot","ID").await);
```
Comment (dendisuhubdy) [inline rust/src/main.rs]: "crypto" is product
"spot" is market 

for example https://api.bitwyre.com/public/pairs?market=crypto&product=spot&country=ID

Comment (dendisuhubdy) [reply rust/src/main.rs]: make both product and market parameter inputs @agustinustheo

File: rust/src/main.rs [hunk 2]
```diff
@@ -9,134 +9,38 @@ async fn async_main() {
...
+    println!("{:?}", public::get_price_index_async(instrument, None, None, None, None).await);
+    println!("{:?}", public::get_search_async("ID", None).await);
+    println!("{:?}", public::get_matching_engine_order_lag_async(None).await);
+    println!("{:?}", public::get_matching_engine_throughput_async(None).await);
+    println!("{:?}", private::get_account_balance_async().await);
+    println!("{:?}", private::get_account_statement_async().await);
+    println!("{:?}", private::get_transaction_histories_async().await);
+    println!("{:?}", private::get_open_orders_async(instrument, 0, 0).await);
+    println!("{:?}", private::get_closed_orders_async(instrument, 0, 0).await);
+    println!("{:?}", private::get_order_info_async("3f128876-a082-4b18-8fd4-abbcc4aa7915").await);
```
Comment (dendisuhubdy) [inline rust/src/main.rs]: "3f128876-a082-4b18-8fd4-abbcc4aa7915" is order ID

that order ID is not general, make it a parameter

File: rust/src/main.rs [hunk 3]
```diff
@@ -9,134 +9,38 @@ async fn async_main() {
...
+    println!("{:?}", private::get_trade_history_async(instrument, 2, 0, 0).await);
+    println!("{:?}", private::cancelling_open_order_per_instrument_async(instrument).await);
+    println!("{:?}", private::cancelling_open_order_per_orderids_async(vec!["5033216a-dad5-4cb6-87d6-0084603be699", "3d19d8d1-c576-4109-8671-5e4115cf6e5e"], vec![-1, -1]).await);
```
Comment (dendisuhubdy) [inline rust/src/main.rs]: this too

File: rust/src/main.rs [hunk 4]
```diff
@@ -9,134 +9,38 @@ async fn async_main() {
...
+    println!("{:?}", private::opening_new_order_async(instrument, 1, Some(35000.to_string()), 2, "0.0001").await);
```
Comment (dendisuhubdy) [inline rust/src/main.rs]: adjust as input parameters

File: rust/src/main.rs [hunk 5]
```diff
@@ -9,134 +9,38 @@ async fn async_main() {
...
+    println!("{:?}", private::opening_new_order_async(instrument, 1, Some("".to_string()), 1, "0.0001").await);
 }
```
Comment (dendisuhubdy) [inline rust/src/main.rs:53]: you are missing cancel order @agustinustheo

**Review by dendisuhubdy — CHANGES_REQUESTED** [review]

File: rust/src/main.rs (hunk 5 above)
Comment (thomsoe84) [reply rust/src/main.rs:53]: i think cancel order is in line 40-41 mas @dendisuhubdy

**Review by dendisuhubdy — APPROVED** [review]

**Pull request merged**